mypy failing with latest google-cloud-core

```
nox > mypy google/cloud
google/cloud/bigquery/client.py:209: error: Incompatible types in assignment (expression has type "Tuple[str, str]", base class "Client" defined the type as "Optional[Tuple[str]]")
google/cloud/bigquery/_http.py:55: error: Incompatible types in assignment (expression has type "str", base class "JSONConnection" defined the type as "None")
google/cloud/bigquery/_http.py:58: error: Incompatible types in assignment (expression has type "str", base class "JSONConnection" defined the type as "None")
Found 3 errors in 2 files (checked 49 source files)
nox > Command mypy google/cloud failed with exit code 1
```

https://github.com/googleapis/python-bigquery/pull/1056#issuecomment-965005522

--- a/google/cloud/_http/__init__.py
+++ b/google/cloud/_http/__init__.py
@@ -19,6 +19,7 @@ import collections.abc
 import json
 import os
 import platform
+from typing import Optional
 from urllib.parse import urlencode
 import warnings
 
@@ -168,19 +169,19 @@ class JSONConnection(Connection):
     must be updated by subclasses.
     """
 
-    API_BASE_URL = None
+    API_BASE_URL: Optional[str] = None
     """The base of the API call URL."""
 
-    API_BASE_MTLS_URL = None
+    API_BASE_MTLS_URL: Optional[str] = None
     """The base of the API call URL for mutual TLS."""
 
     ALLOW_AUTO_SWITCH_TO_MTLS_URL = False
     """Indicates if auto switch to mTLS url is allowed."""
 
-    API_VERSION = None
+    API_VERSION: Optional[str] = None
     """The version of the API, used in building the API call's URL."""
 
-    API_URL_TEMPLATE = None
+    API_URL_TEMPLATE: Optional[str] = None
     """A template for the URL of a particular API call."""
 
     def get_api_base_url_for_mtls(self, api_base_url=None):

--- a/google/cloud/client/__init__.py
+++ b/google/cloud/client/__init__.py
@@ -144,7 +144,7 @@ class Client(_ClientFactoryMixin):
             to acquire default credentials.
     """
 
-    SCOPE: Union[Tuple[str], None] = None
+    SCOPE: Union[Tuple[str, ...], None] = None
     """The scopes required for authenticating with a service.
 
     Needs to be set by subclasses.

